anasymod/templates/xsct_program.py

```diff
@@ -2,34 +2,83 @@
 # 1. https://www.xilinx.com/html_docs/xilinx2018_1/SDK_Doc/xsct/use_cases/xsdb_standalone_app_debug.html
 # 2. https://github.com/Digilent/Arty-Z7-20-linux_bd/blob/master/sdk/.sdk/launch_scripts/xilinx_c-c%2B%2B_application_(system_debugger)/system_debugger_using_debug_video.elf_on_local.tcl
 # 3. https://xilinx-wiki.atlassian.net/wiki/spaces/A/pages/18842240/Programming+QSPI+from+U-boot+ZC702
+# 4. https://github.com/analogdevicesinc/no-OS/blob/master/scripts/xsdb.tcl
+# 5. https://www.xilinx.com/html_docs/xilinx2018_1/SDK_Doc/xsct/use_cases/xsdb_debug_app_zynqmp.html
+
+from pathlib import Path
 
 class TemplXSCTProgram:
-    def __init__(self, sdk_path, bit_path, hw_path, tcl_path, cpu_filter='"ARM*#0"',
-                 sw_name='sw', program_fpga=True, reset_system=True):
+    def __init__(self, sdk_path, bit_path, hw_path, tcl_path,
+                 sw_name='sw', program_fpga=True, reset_system=True,
+                 loadhw=True, init_cpu=True, download=True,
+                 run=True, connect=True, is_ultrascale=False,
+                 xsct_install_dir=None):
+
+        # save settings
+        self.sdk_path = sdk_path
+        self.bit_path = bit_path
+        self.hw_path = hw_path
+        self.tcl_path = tcl_path
+        self.sw_name = sw_name
+        self.is_ultrascale = is_ultrascale
+        self.xsct_install_dir = xsct_install_dir
 
         # initialize text
         self.text = ''
 
-        # apply commands
+        ##################
+        # apply commands #
+        ##################
 
-        self.connect()
+        # connect
+        if connect:
+            self.connect()
 
-        self.select_cpu(cpu_filter)
+        # load utility functions for Zynq MP
```

Comment:
not sure if this is really necessary, but it's what Vitis does as part of the GUI programming flow